lib/Service/SocialApiService.php

```diff
@@ -0,0 +1,206 @@
+<?php
+/**
+ * @copyright Copyright (c) 2020 Matthias Heinisch <nextcloud@matthiasheinisch.de>
+ *
+ * @author Matthias Heinisch <nextcloud@matthiasheinisch.de>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Contacts\Service;
+
+use OCA\Contacts\Service\Social\CompositeSocialProvider;
+
+use OCP\Contacts\IManager;
+use OCP\IAddressBook;
+
+use OCP\IConfig;
+use OCP\AppFramework\Http;
+use OCP\AppFramework\Http\JSONResponse;
+
+
+class SocialApiService {
+
+	protected $appName;
+
+	/** @var CompositeSocialProvider */
+	private $socialProvider;
+	/** @var IManager */
+	private  $manager;
+	/** @var IConfig */
+	private  $config;
+
+	public function __construct(string $AppName,
+					CompositeSocialProvider $socialProvider,
+					IManager $manager,
+					IConfig $config) {
+
+		$this->appName = $AppName;
+		$this->socialProvider = $socialProvider;
+		$this->manager = $manager;
+		$this->config = $config;
+	}
+
+
+	/**
+	 * @NoAdminRequired
+	 *
+	 * returns an array of supported social networks
+	 *
+	 * @returns {array} array of the supported social networks
+	 */
+	public function getSupportedNetworks() : array {
+		$isAdminEnabled = $this->config->getAppValue($this->appName, 'allowSocialSync', 'yes');
+		if ($isAdminEnabled !== 'yes') {
+			return array();
+		}
+		return $this->socialProvider->getSupportedNetworks();
+	}
+
+
+	/**
+	 * @NoAdminRequired
+	 *
+	 * Creates the photo start tag for the vCard
+	 *
+	 * @param {float} version the version of the vCard
+	 * @param {array} header the http response headers containing the image type
+	 *
+	 * @returns {String} the photo start tag or null in case of errors
+	 */
+	protected function getPhotoTag(float $version, array $header) : ?string {
+
+		$type = null;
+
+		// get image type from headers
+		foreach ($header as $value) {
+			if (preg_match('/^Content-Type:/i', $value)) {
+				if (stripos($value, "image") !== false) {
+					$type = substr($value, stripos($value, "image"));
+				}
+			}
+		}
+		if (is_null($type)) {
+			return null;
+		}
+
+		// return respective photo tag
+		if ($version >= 4.0) {
+			return "data:" . $type . ";base64,";
+		}
+
+		if ($version >= 3.0) {
+			$type = str_replace('image/', '', $type);
+			return "ENCODING=b;TYPE=" . strtoupper($type) . ":";
```

Comment:
How come this is not working? See also: https://github.com/nextcloud/contacts/blob/070fc6c8e3e47cc08e0941292569e0e121585453/src/components/ContactDetails/ContactDetailsAvatar.vue#L258-L267

Reply:
I found out that I can add a photo vor version 3.0 by using `$changes['PHOTO;ENCODING=b;TYPE=jpeg;VALUE=BINARY'] = base64_encode($socialdata);`, but it doesn't overwrite the previous `PHOTO;ENCODING=b;TYPE=jpeg;VALUE=BINARY` from the contact, so it only shows up after one deletes the current one.

Reply:
I could not figure out how to update that properly, so I am explicitly overwriting the previous photo now.